### PR TITLE
check for existing invite username

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR exists for invite username.

### CODE REVIEW

File: skyportal/onboarding.py [hunk 1]
```diff
@@ -26,7 +26,9 @@ def create_user(strategy, details, backend, uid, user=None, *args, **kwargs):
     invite_token = strategy.session_get("invite_token")
 
     try:
-        existing_user = DBSession().query(User).filter(User.oauth_uid == uid).first()
+        existing_user = (
+            DBSession().scalars(sa.select(User).where(User.oauth_uid == uid)).first()
```
Comment (Theodlz) [inline skyportal/onboarding.py:30]: please replace with .scalar(), rather than `scalars().first()`.

File: skyportal/onboarding.py [hunk 2]
```diff
@@ -45,10 +47,10 @@ def create_user(strategy, details, backend, uid, user=None, *args, **kwargs):
 
             invitation = (
                 DBSession()
-                .query(Invitation)
-                .filter(Invitation.token == invite_token)
+                .scalars(sa.select(Invitation).where(Invitation.token == invite_token))
```
Comment (Theodlz) [inline skyportal/onboarding.py:50]: same comment here

File: skyportal/onboarding.py [hunk 3]
```diff
@@ -62,6 +64,14 @@ def create_user(strategy, details, backend, uid, user=None, *args, **kwargs):
                     "Authentication Error: Invite token has already been used."
                 )
 
+            check_username = (
+                DBSession()
```
Comment (Theodlz) [inline skyportal/onboarding.py:68]: same comment here

**Review by Theodlz — APPROVED** [review]
LGTM, minor comments just to be consistent with the sqlalchemy 2 syntax from baselayer.

File: skyportal/onboarding.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/onboarding.py]: Done.

File: skyportal/onboarding.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/onboarding.py]: Done.

**Pull request merged**